### ISSUE TITLE
Git clone issues

### ISSUE THREAD

**Issue by markhowells** [chadparry/kodi-repository.chad.parry.org#2]
I'm getting an error trying to make a repo for a bitbucket hosted repo.  Works great for a locally held addon. 

OS is ubuntu 16.04 and I using python 2.7.12

```
root@manage:~/repo#` ./create_repository.py https://seaspray@bitbucket.org/unblockr/addon.git
Traceback (most recent call last):
  File "./create_repository.py", line 391, in <module>
    main()
  File "./create_repository.py", line 387, in main
    args.addon, data_path, info_path, checksum_path, args.compressed)
  File "./create_repository.py", line 317, in create_repository
    raise result.exc_info[1]
TypeError: Expected bytes or text, but got <open file './script.service.unblockr/script.service.unblockr-0.1.2.zip', mode 'wb' at 0x7f0541eb99c0>
```

The zip zile is created (mode 644) with zero bytes ...

**Comment by chadparry** [comment]
It looks like a bad pathspec was being passed to the GitPython tool. I fixed it in 2e53cc7, v1.3.1.
I had been able to reproduce an error with your command, but it was different error message than the one you showed. That means that there is possibly a different bug still lurking. Please just reopen this issue if it isn't all fixed, and I'll collect some more information from you.

**Comment by markhowells** [comment]
I downloaded 1.3.1 as you suggested. 

Sadly, I'm seeing the same error

```
root@manage:~/repo# wget https://raw.githubusercontent.com/chadparry/kodi-repository.chad.parry.org/tools-1.3.1/tools/create_repository.py
--2017-05-09 08:09:14--  https://raw.githubusercontent.com/chadparry/kodi-repository.chad.parry.org/tools-1.3.1/tools/create_repository.py
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.60.133
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.60.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 14022 (14K) [text/plain]
Saving to: ‘create_repository.py.1’
create_repository.py.1                                   100%[==================================================================================================================================>]  13.69K  --.-KB/s    in 0.001s
2017-05-09 08:09:15 (9.71 MB/s) - ‘create_repository.py.1’ saved [14022/14022]
root@manage:~/repo# chmod a+x create_repository.py
root@manage:~/repo# ./create_repository.py https://seaspray@bitbucket.org/unblockr/addon.git
Traceback (most recent call last):
  File "./create_repository.py", line 391, in <module>
    main()
  File "./create_repository.py", line 387, in main
    args.addon, data_path, info_path, checksum_path, args.compressed)
  File "./create_repository.py", line 317, in create_repository
    raise result.exc_info[1]
TypeError: Expected bytes or text, but got <open file './script.service.unblockr/script.service.unblockr-0.1.2.zip', mode 'wb' at 0x7fda6f5219c0>
root@manage:~/repo#
```
Specifying a full set of repo parameters 
```
./create_repository.py https://seaspray@bitbucket.org/unblockr/addon.git#master:.
```
also results in the same error. Debugging the script around line 150 with
```
    print 'folder=', clone_folder, '\n'
    print 'repo=', clone_repo, '\n'
    print 'branch=', clone_branch, '\n'
    print 'path=', clone_path, '\n'
```
together with the value of 'cloned' yields nothing untoward looking (to my eyes at least)
```
root@manage:~/repo# ./create_repository.py https://seaspray@bitbucket.org/unblockr/addon.git#master:.
folder= /tmp/tmpJSweo7repo-
repo= https://seaspray@bitbucket.org/unblockr/addon.git
branch= master
path= .
cloned= <git.Repo "/tmp/tmpJSweo7repo-/.git">
```

**Comment by chadparry** [comment]
Thanks for the debugging info. Could you please clarify one part of that 
output?

> |Saving to: ‘create_repository.py.1’|
Did you rename this from create_repository.py.1 to create_repository.py? 
The wget command had renamed it because it saw that create_repository.py 
already existed. There's a chance that you ran those commands on the old 
1.3.0 version.

**Comment by markhowells** [comment]
Hi Chad,

You're correct :(  And I had noticed. In fact about 5 mins after posting
the comment I deleted my comment but I guess the email note to you had
already been dispatched. Embarassing really, to make such a rookie error.

Thank you for your work with this tho.. I had (and still do) find much of
the Kodi documentation around addon development highly confusing, outdated
and contradictory. Your work is not only productive in automating my
deployments, but also helped me sort out the precise process required when
pushing a new release. Prior to using this tool, it was a distinctly scary,
approximate affair....

Thanks again.

Cheers,

Mark
